text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,337 @@
+
+All `@aws-cdk/` namespaced libraries will continue to be published with current `1.x.x` versions.
+
+### v1 Deprecation
+
+`@aws-cdk/` namespaced modules will continue to receive non-breaking changes and be released regularly.
+
+When v1 module support is discontinued will ultimately be dictated by the introduction of some feature or change that the team agrees is worth the cost of dropping support and requires us to do so. Alternatively, if the burden of maintaining v1 module support is no longer warranted by the number of users continuing to rely on them.
+
+## Major Version Rollout Tooling
+
+To aid in this phased release, some features will need to be added to JSII and other cdk dev tools. These features will help us with rolling out major versions in the future. Whenever a new breaking release occurs, deprecated APIs will be hidden and feature flags will be reset. When we are going through the next cycle of breaking changes, in this case v3.0, then code related to deprecated APIs and feature flags introduced during the preceding major version cycle, v1.0 in this case, can be removed since they no longer will be supported.
+
+#### [JSII Prerelease Versioning Awareness]()
+
+In order to support releasing the `aws-cdk-lib` with a prerelease version number, JSII needs to parse the version number of the module its building and recognize it as a "prerelease" and convert it to whatever format is equivalent in the target languages package manager.
+
+#### [JSII Submodule Support](https://github.com/aws/jsii/issues/1286)
+
+JSII currently doesn't handle namespacing of exports. Support for this will be needed to allow building the `aws-cdk-lib` module in all of the cdk supported target languages. Though the initial work is complete, some open questions remain about the scope of this feature. Namely, in order to make migrating to the new module structure as easy as possible, [namespace renaming]() could potentially allow C#, Python, and Java users to not have to change the structure of their import statements.
+
+#### [`aws-cdk-lib` Version Unlocking]()
+
+Since `aws-cdk-lib` will have a version number differing from the one used by lerna to version all of the other packages in the repo until the "stable" phase, changes to the align-version and bump scripts will have to be made in the repo to handle this.
+
+#### [Deprecation Aware Builds]()
+
+A strategy for selectively including or excluding apis marked `deprecated` during build time. This can be performed at the JSII level and potentially can be linked to specific version numbers with additional information in the `deprecated` annotation. JSII can warn when APIs marked for deprecation within a specific major version release are being included in the build. For example:
```

Comment:
What about deprecated packages?

Reply:
Ideally we should omit them from the monocdk, but I think it might be impossible given we have some stable APIs that depend on deprecated/experimental APIs.